app/src/main/java/com/example/sharewhatyoucanproject/HomeFragment.kt

```diff
@@ -0,0 +1,133 @@
+package com.example.sharewhatyoucanproject
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.os.Looper
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import com.example.sharewhatyoucanproject.DonorActivity
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.GeoPoint
+import com.google.type.LatLng
+
+class HomeFragment : Fragment() {
+    var uploadlayout: Button? = null
+    var nameofuser: TextView? = null
+    lateinit var checklocation: Button
+    var latitude: Double = 44.5581222
+
+    // -123.2734361
+    var longitute: Double = -123.2734361
+    var mytest: LatLng? = null
+    var test: Long = 0L
+    lateinit var convertbtn: Button
+    lateinit var locationRequest: com.google.android.gms.location.LocationRequest
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_home, container, false)
+        uploadlayout = view.findViewById(R.id.uploadlayout)
+        val nameofuser = view.findViewById<TextView>(R.id.nameofuser)
+        nameofuser.text = "" + FirebaseAuth.getInstance().currentUser!!.displayName
+        checklocation = view.findViewById(R.id.checklocation)
+
+        convertbtn = view.findViewById(R.id.convertbtn)
+        locationRequest = com.google.android.gms.location.LocationRequest.create()
+        checklocation.setOnClickListener(View.OnClickListener {
+
+            if (context?.let { it1 ->
+                    ActivityCompat.checkSelfPermission(
+                        it1,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    )
+                } != PackageManager.PERMISSION_GRANTED && context?.let { it1 ->
+                    ActivityCompat.checkSelfPermission(
+                        it1,
+                        Manifest.permission.ACCESS_COARSE_LOCATION
+                    )
+                } != PackageManager.PERMISSION_GRANTED
+            ) {
```

Comment:
As a non-Android developer, I don't understand what this says.  Is this an Android/Kotlin pattern?  If not, is there a way to write this that would be more accessible?